### PR TITLE
Have nodeup retry kops-controller bootstrapping sooner if DNS isn't setup

### PR DESCRIPTION
**Pull request by rifelpet**
ref: https://github.com/kubernetes/kops/issues/11050#issuecomment-803620244

### CODE REVIEW

**Review by olemarkus — APPROVED** [review]

**Comment by rifelpet** [comment]
hm looks like these errors are being hit over 10 minutes after the dns record is created. Will need some additional troubleshooting

**Comment by justinsb** [comment]
Yeah, not sure what's going on there but I see the same thing.  Maybe log the error and the host name we're resolving?

File: upup/pkg/fi/nodeup/nodetasks/bootstrap_client.go [hunk 1]
```diff
@@ -155,6 +159,19 @@ func (b *KopsBootstrapClient) QueryBootstrap(ctx context.Context, req *nodeup.Bo
...
 
+	if ips, err := net.LookupIP(b.BaseURL.Host); err != nil {
+		if dnsErr, ok := err.(*net.DNSError); ok && dnsErr.IsNotFound {
+			return nil, fi.NewTryAgainLaterError("kops-controller DNS not setup yet")
+		}
+		return nil, err
+	} else {
+		for _, ip := range ips {
+			if ip.String() == cloudup.PlaceholderIP {
+				return nil, fi.NewTryAgainLaterError("kops-controller DNS not setup yet")
```
Comment (justinsb) [inline upup/pkg/fi/nodeup/nodetasks/bootstrap_client.go:170]: Probably also a good idea to differentiate the two messages e.g. ("kops-controller DNS not setup yet (placeholder IP found)" vs "kops-controller DNS not setup yet (error was %v)"

Comment (rifelpet) [reply upup/pkg/fi/nodeup/nodetasks/bootstrap_client.go]: Good idea, it turns out `URL.Host` includes the port number which will never resolve. Updating it to `URL.Hostname()` seems to have shaved off a couple minutes on the `kops validate cluster` time in this small sample size of presubmit jobs.

**Review by olemarkus — APPROVED** [review]

**Pull request merged**